.planning/codebase/TESTING.md

```
@@ -1,193 +1,183 @@
+
+**Location:**
+- BATS: dedicated `tests/` directory at repo root (not co-located with shell sources)
+- TypeScript: `tests/code-taste.test.ts` alongside `lib/code-taste/` implementation
 
-### Directory Layout
+**Naming:**
+- `*.bats` for shell tests; `pr_<feature>.bats` for config contract tests per tool/PR
+- `*.test.ts` for Bun unit tests
 
+**Structure:**
 ```
```

markdownlint-cli2 (0.23.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)